packages/interactive_media_ads/lib/src/platform_interface/platform_ads_rendering_settings.dart

```diff
@@ -0,0 +1,134 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+import 'interactive_media_ads_platform.dart';
+import 'ui_element.dart';
+
+/// Object specifying creation parameters for creating a
+/// [PlatformAdsRenderingSettings].
+///
+/// Platform specific implementations can add additional fields by extending
+/// this class.
+///
+/// This example demonstrates how to extend the
+/// [PlatformAdsRenderingSettingsCreationParams] to provide additional platform
+/// specific parameters.
+///
+/// When extending [PlatformAdsRenderingSettingsCreationParams], additional
+/// parameters should always accept `null` or have a default value to prevent
+/// breaking changes.
+///
+/// ```dart
+/// final class AndroidPlatformAdsRenderingSettingsCreationParams
+///     extends PlatformAdsRenderingSettingsCreationParams {
+///   const AndroidPlatformAdsRenderingSettingsCreationParams({
+///     super.bitrate,
+///     this.enableCustomTabs,
+///   }) : super();
+///
+///   factory AndroidPlatformAdsRenderingSettingsCreationParams.fromPlatformAdsRenderingSettingsCreationParams(
+///     PlatformAdsRenderingSettingsCreationParams params, {
+///     bool? enableCustomTabs,
+///   }) {
+///     return AndroidPlatformAdsRenderingSettingsCreationParams(
+///       bitrate: params.bitrate,
+///       enableCustomTabs: enableCustomTabs,
+///     );
+///   }
+///
+///   final bool? enableCustomTabs;
+/// }
+/// ```
+@immutable
+base class PlatformAdsRenderingSettingsCreationParams {
+  /// Used by the platform implementation to create a new
+  /// [PlatformAdsRenderingSettings].
+  const PlatformAdsRenderingSettingsCreationParams({
+    this.bitrate,
+    this.enablePreloading,
+    this.loadVideoTimeout,
+    this.mimeTypes,
+    this.playAdsAfterTime,
+    this.uiElements,
+  });
+
+  /// Maximum recommended bitrate.
+  ///
+  /// The value is in kbit/s.
+  ///
+  /// The SDK will select media which has a bitrate below the specified max or
+  /// the closest bitrate if there is no media with a lower bitrate found.
+  ///
+  /// If null, the bitrate will be selected by the SDK, using the currently
+  /// detected network speed (cellular or Wi-Fi).
+  final int? bitrate;
+
+  /// If set, the SDK will instruct the player to load the creative in response
+  /// initializing the ads manager.
+  ///
+  /// This allows the player to preload the ad at any point before starting the
+  /// ads manager.
+  ///
+  /// If null, the platform will decide the default value.
+  final bool? enablePreloading;
```

Comment:
Note: This value defaults to `false` on Android and `true` on iOS. To prevent a breaking change, I made the value nullable to indicate the platform will handle the default. Same is true for the other fields.